src/wallet/rpc/spend.cpp

```diff
@@ -88,8 +88,21 @@ std::set<int> InterpretSubtractFeeFromOutputInstructions(const UniValue& sffo_in
     return sffo_set;
 }
 
-static UniValue FinishTransaction(const std::shared_ptr<CWallet> pwallet, const UniValue& options, const CMutableTransaction& rawTx)
+static UniValue FinishTransaction(const std::shared_ptr<CWallet> pwallet, const UniValue& options, CMutableTransaction& rawTx)
 {
+    bool can_anti_fee_snipe = !options.exists("locktime");
+
+    for (const CTxIn& tx_in : rawTx.vin) {
+        // Can not be FINAL for locktime to work
+        can_anti_fee_snipe &= tx_in.nSequence != CTxIn::SEQUENCE_FINAL;
+    }
+
+    if (can_anti_fee_snipe) {
+        LOCK(pwallet->cs_wallet);
```

Comment:
I'm not clear why do we need a wallet lock here?

Reply:
Here the wallet is locked because it runs `CWallet::GetLastBlockHash` and `CWallet::GetLastBlockHeight` which both require `pwallet->cs_wallet` to be locked.